src/main/java/com/google/javascript/cl2dts/DeclarationGenerator.java

```diff
@@ -647,6 +647,9 @@ public Void caseTemplatizedType(TemplatizedType type) {
             emit("NodeList");
             return null;
           }
+          if (type.getDisplayName().equals("IThenable")) {
```

Comment:
Java 7 allows string literals in switch statements:
http://docs.oracle.com/javase/7/docs/technotes/guides/language/strings-switch.html

Would it make sense to tackle these as cases based on display name?
```java
switch (type.getDisplayName()) {
  case "Arguments":
  case "NewArguments":
    ...
```